src/App.tsx

```diff
@@ -12,212 +35,56 @@ export const App: React.FC = () => {
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    if (user) {
+      getTodos(user.id)
+        .then((todosFromServer) => setTodos(todosFromServer))
+        .catch(() => setErrorMessage('Todos loading failed'));
+    }
   }, []);
 
+  const activeTodos = useMemo(() => (
+    todos.filter(todo => !todo.completed)
+  ), [todos, filterType]);
+
+  const visibleTodos = useMemo(() => (
+    todos.filter(todo => {
+      switch (filterType) {
+        case FilterType.All:
+          return todo;
+
+        case FilterType.Active:
+          return !todo.completed;
```

Comment:
Use the switch once, before filtering
Cause in your case you are checking filter type during the iteration many times

```suggestion
    switch (filterType) {
        case FilterType.Active:
          return todos.filter(todo => (
            !todo.completed
          )
          
        case .... and so on
```